components/global/test_helpers.tsx

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {ProductComponent} from 'types/store/plugins';
+
+type TestProductComponent = Omit<ProductComponent, 'switcherIcon'> & {
+    switcherIcon: string;
+}
+
+export function makeProduct(name: string): TestProductComponent {
+    return {
+        id: name,
+        pluginId: '',
+        switcherIcon: `product-${name.toLowerCase()}`,
```

Comment:
Instead of having `makeProduct` return a new type and then using `as ProductComponent` everywhere else, what if we just change this line to
```suggestion
        switcherIcon: `product-${name.toLowerCase()}` as ProductComponent['switcherIcon'],
```
to fake that this value is the correct type here instead?